src/Version.php

```diff
@@ -1,226 +1,162 @@
+        return $this->parts[2] != '0';
     }
 
     /**
-     * Get the latest version from Github
+     * Is there an extension?
      *
-     * @param  Http\Client $httpClient Configured HTTP client
-     * @return string|null API response or false on error
+     * @param  string|null $extension a version extension
+     * @return bool
      */
-    protected static function getLatestFromGithub(Http\Client $httpClient = null)
+    public function hasExtension($extension = null)
     {
-        $url = 'https://api.github.com/repos/zendframework/zf2/git/refs/tags/release-';
-
-        if ($httpClient === null) {
-            $context = stream_context_create(
-                [
-                    'http' => [
-                        'user_agent' => sprintf('Zend-Version/%s', self::VERSION),
-                    ],
-                ]
-            );
-            $apiResponse = file_get_contents($url, false, $context);
-        } else {
-            $request = new Http\Request();
-            $request->setUri($url);
-            $httpClient->setRequest($request);
-            $apiResponse = self::getApiResponse($httpClient);
-        }
-
-        if (!$apiResponse) {
+        if (! isset($this->parts[3])) {
             return false;
         }
-
-        $decodedResponse = Json::decode($apiResponse, Json::TYPE_ARRAY);
-
-        // Simplify the API response into a simple array of version numbers
-        $tags = array_map(function ($tag) {
-            return substr($tag['ref'], 18); // Reliable because we're
-                                            // filtering on 'refs/tags/release-'
-        }, $decodedResponse);
-
-        // Fetch the latest version number from the array
-        return array_reduce($tags, function ($a, $b) {
-            return version_compare($a, $b, '>') ? $a : $b;
-        });
+        return ($extension !== null) ? $this->parts[3] === $extension : true;
     }
 
     /**
-     * Get the latest version from framework.zend.com
+     * Compare another version.
      *
-     * @param  Http\Client $httpClient Configured HTTP client
-     * @return string|null API response or false on error
+     * @link   http://php.net/manual/function.version-compare.php
+     * @param  string $another  another version to compare
+     * @param  string $operator a comparison operator
+     * @return bool
      */
-    protected static function getLatestFromZend(Http\Client $httpClient = null)
+    public function compare($another, $operator = '>=')
     {
-        $url = 'http://framework.zend.com/api/zf-version?v=2';
-
-        if ($httpClient === null) {
-            $apiResponse = file_get_contents($url);
-        } else {
-            $request = new Http\Request();
-            $request->setUri($url);
-            $httpClient->setRequest($request);
-            $apiResponse = self::getApiResponse($httpClient);
-        }
+        return version_compare((string) $another, $this->version, strtolower($operator));
```

Comment:
Oops.  It's only an `int` if the operator is not used.
> When using the optional operator argument, the function will return TRUE if the relationship is the one specified by the operator, FALSE otherwise.